Does not build with Ninja on linux

Using these cmake options an d cmake 3.23.0:
```
-G Ninja  -DWITH_DBUS=ON \
              -DWITH_DOTNET=OFF \
              -DWITH_GNOME3=ON \
              -DWITH_KDE=ON \
              -DWITH_MOZJS=OFF \
              -DWITH_NATUS=OFF \
              -DWITH_NM=OFF \
              -DWITH_PERL=ON \
              -DWITH_PYTHON2=ON \
              -DWITH_PYTHON3=ON \
              -DWITH_VALA=ON \
              -DWITH_WEBKIT=OFF \
              -DWITH_WEBKIT3=ON \
              -DPERL_VENDORINSTALL=ON \
              -DBIPR=OFF \
              -DBUILD_TESTING=OFF \
              -Wno-dev .."
```
This is the error I see:

```
Building from source, this may take a while...
-- The C compiler identification is GNU 11.2.1
-- The CXX compiler identification is GNU 11.2.1
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/local/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/local/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found Python2Interp: /usr/local/bin/python (found version "3.10.3") 
-- Found python 2 version 3.10
<string>:1: DeprecationWarning: The distutils package is deprecated and slated for removal in Python 3.12. Use setuptools or check PEP 632 for potential alternatives
<string>:1: DeprecationWarning: The distutils.sysconfig module is deprecated, use sysconfig instead
-- Using PYTHON2_SITEPKG_DIR=/usr/local/lib/python3.10/site-packages
-- Found Python3Interp: /usr/local/bin/python3.10  
-- Found python 3 version 3.10
<string>:1: DeprecationWarning: The distutils package is deprecated and slated for removal in Python 3.12. Use setuptools or check PEP 632 for potential alternatives
<string>:1: DeprecationWarning: The distutils.sysconfig module is deprecated, use sysconfig instead
-- Using PYTHON3_SITEPKG_DIR=/usr/local/lib/python3.10/site-packages
-- Found Perl: /usr/local/bin/perl (found version "5.34.1") 
-- Found PerlLibs: /usr/local/lib64/perl5/5.34.1/core_perl/CORE/libperl.so (found version "5.34.1") 
-- Found PkgConfig: /usr/local/bin/pkg-config (found version "0.29.2") 
-- Checking for module 'dbus-1'
--   Found dbus-1, version 1.13.18
-- Checking for modules 'gio-2.0>=2.26;gobject-2.0'
--   Found gio-2.0, version 2.71.3
--   Found gobject-2.0, version 2.71.3
-- Checking for modules 'NetworkManager;dbus-1'
--   No package 'NetworkManager' found
-- Checking for module 'duktape'
--   Found duktape, version 2.6.0
MODULES TO BUILD:
	m	config_pacrunner
	*	config_envvar
		config_sysconfig
		config_gnome
	m	config_gnome3
	m	config_kde
		config_macosx
		config_w32reg
	*	ignore_domain
	*	ignore_hostname
	*	ignore_ip
		network_networkmanager
		pacrunner_mozjs
		pacrunner_natus
		pacrunner_webkit
	m	pacrunner_duktape
	*	wpad_dns_alias
1 pacrunner(s) are enabled

-- Configuring done
-- Generating done
CMake Error:
  Running

   '/usr/local/bin/ninja' '-C' '/usr/local/tmp/crew/libproxy.20220330030130.dir/builddir' '-t' 'recompact'

  failed with:

   ninja: error: build.ninja:1716: multiple rules generate test [-w dupbuild=err]
```

--- a/bindings/perl/lib/CMakeLists.txt
+++ b/bindings/perl/lib/CMakeLists.txt
@@ -1,2 +1,2 @@
-add_custom_target(PMlibproxy ALL ${CMAKE_COMMAND} -E copy ${CMAKE_CURRENT_SOURCE_DIR}/Libproxy.pm ${CMAKE_BINARY_DIR}/perl/blib/lib/Libproxy.pm)
+add_custom_target(PMlibproxy ALL ${CMAKE_COMMAND} -E copy ${CMAKE_CURRENT_SOURCE_DIR}/Libproxy.pm ${CMAKE_BINARY_DIR}/perl/Net/Libproxy.pm)
 install( FILES Libproxy.pm DESTINATION ${PX_PERL_ARCH}/Net )

--- a/bindings/perl/src/CMakeLists.txt
+++ b/bindings/perl/src/CMakeLists.txt
@@ -9,7 +9,7 @@ add_custom_command(OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/Libproxy.c
 
 set(Libproxy_LIB_SRCS Libproxy.c)
 
-set(LIBRARY_OUTPUT_PATH ${CMAKE_BINARY_DIR}/perl/blib/arch/auto/Net)
+set(LIBRARY_OUTPUT_PATH ${CMAKE_BINARY_DIR}/perl/auto/Net/Libproxy)
 add_library(PLlibproxy SHARED ${Libproxy_LIB_SRCS})
 
 set(PLlibproxy_LIB_DEPENDENCIES libproxy pthread)

--- a/bindings/perl/t/CMakeLists.txt
+++ b/bindings/perl/t/CMakeLists.txt
@@ -1,1 +1,2 @@
-add_custom_target(test prove -b ${CMAKE_CURRENT_SOURCE_DIR})
+add_test(NAME perl COMMAND prove -b ${CMAKE_CURRENT_SOURCE_DIR})
+set_property(TEST perl APPEND PROPERTY ENVIRONMENT "PERL5LIB=${CMAKE_BINARY_DIR}/perl")

